Added tests for Burger, Bun, Ingredient and IngredientType classes

ncherny opened the pull request.

ncherny committed:
Added tests for Burger, Bun, Ingredient and IngredientType classes

okryukova commented on src/test/java/BunTests.java:
```diff
@@ -0,0 +1,31 @@
+public class BunTests {
+    private Bun bun;
+
+    private static float EXPECTED_PRICE = 10.5f;
```
⛔️Нужно исправить.  Для константы пропущено ключевое слово final. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/BurgerIngredientsTests.java:
```diff
@@ -0,0 +1,45 @@
+    @Test
+    public void moveIngredientPositiveTest() {
+        burger = new Burger();
+        Ingredient ingredient1 = new Ingredient(INGREDIENT_DEFAULT_TYPE, INGREDIENT_DEFAULT_NAME, INGREDIENT_DEFAULT_PRICE);
```
⛔️Нужно исправить.  Нужно использовать моки

okryukova commented on src/test/java/BurgerIngredientsTests.java:
```diff
@@ -0,0 +1,45 @@
+        burger.addIngredient(ingredient1);
+        burger.addIngredient(ingredient2);
+        burger.moveIngredient(0, 1);
+        assertEquals(burger.ingredients.get(0), ingredient2);
```
⛔️ Нужно исправить. Для юнит-тестов применим подход: один тест, значит одна проверка - один assert. Поправь, пожалуйста этот момент во всем коде

okryukova commented on src/test/java/IngredientTests.java:
⛔️Нужно исправить. enum IngredientType не покрыт тестами.

ncherny committed:
Added fixes for comments

- Fixed some constants missing 'final'
- Updated test for moving ingredient to use mocks
- Separated test for moving ingredient into two to avoid multiple
  assertions
- Added test for enum class
